src/components/main-view/main-view.jsx

```diff
@@ -39,49 +40,104 @@ export const MainView = () => {
-                <Col md={8}>
-                  <MovieView
-                    movie={selectedMovie}
-                    onBackClick={() => setSelectedMovie(null)}
-                  />
-                </Col>
-            ) : movies.length === 0 ? (
-                <div>This list is empty!</div>
-            ) : (
-                <>
-                  {movies.map((movie) => (
-                    <Col className="mb-4" key={movie.id} md={3}>
-                      <MovieCard
-                        movie={movie}
-                        onMovieClick={(newSelectedMovie) => {
-                            setSelectedMovie(newSelectedMovie);
-                        }}
-                      />
-                    </Col>
-                  ))}
-                  <Button variant="primary" type="Logout"
-                    onClick={() => {
-                        setUser(null);
-                        setToken(null);
-                        localStorage.clear();
-                    }}
-                  >
-                    Logout
-                  </Button>
-                </>
-            )}
-        </Row>
+        <BrowserRouter>
+          <NavigationBar
+            user={user}
+            onLoggedOut={() => {
+              setUser(null);
+              setToken(null);
+              localStorage.clear();
+            }}
+          />
+          <Row className="justify-content-center">
+            <Routes>
+              <Route
+                path="/signup"
+                element={
+                  <>
+                    {user ? (
+                      <Navigate to="/" />
+                    ) : (
+                      <Col md={5}>
+                        <SignupView />
+                      </Col>
+                    )}
+                  </>
+                }
+              />
+              <Route
+                path="/login"
+                element={
+                  <>
+                    {user ? (
+                      <Navigate to="/" />
+                    ) : (
+                      <Col md={5}>
+                        <LoginView onLoggedIn={(user, token) => {
+                          setUser(user);
+                          setToken(token);
+                          }} />
+                      </Col>
+                    )}
+                  </>
+                }
+              />
+              <Route
+                path="/profile"
+                element={
+                  <>
+                    {!user ? (
+                      <Navigate to="/login" replace />
+                    ): (
+                      <Col md={8}>
+                        <ProfileView
+                          user={user}
+                          token={token}
+                          movies={movies}
+                        />
+                      </Col>
+                    )}
+                  </>
+                }
+              />
+              <Route
+                path="/movies/movieId"
```

Comment:
path="/movies/:movieId" you are missing the `:` here 

readmore here https://reactrouter.com/en/main/route/route#dynamic-segments